source/Mission.h

```diff
@@ -217,6 +223,13 @@ class Mission {
 	int illegalCargoFine = 0;
 	std::string illegalCargoMessage;
 	bool failIfDiscovered = false;
+	// Parameters for generating random outfit amounts:
+	std::string requestedOutfit;
+	std::string requestedOutfitter;
+	const Outfit *outfit = nullptr;
+	int outfitUnits = 0;
+	int outfitLimit = 0;
+	double outfitProb = 0.;
```

Comment:
I think introducing a helper class here makes sense (to reduce duplication), something like this maybe:

```c++
template <typename T>
struct CargoDefinition { // this name is not that good, but I can't think of another one
    T cargo;
    int count = 0;
    int limit = 0;
    double probability = 0.;
};
```

It would also allow having both a `cargo outfit` and a `cargo outfitter`.

Reply:
This lends it self to generalizing cargo, outfit and outfitter into an array so we could have multiple of all objectives. Could do that as a follow up.